internal/node/node_loader.js

```diff
@@ -71,408 +74,11 @@ node_loader: running ${TARGET} with
-    // On Windows, Bazel sets RUNFILES_MANIFEST_ONLY=1.
-    // On every platform, Bazel also sets RUNFILES_MANIFEST_FILE, but on Linux
-    // and macOS it's faster to use the symlinks in RUNFILES_DIR rather than resolve
-    // through the indirection of the manifest file.
-    // We also need to construct a reverse map to resolve relative files from existing
-    // manifest entries.
-    process.env.RUNFILES_MANIFEST_ONLY === '1' &&
-    loadRunfilesManifest(process.env.RUNFILES_MANIFEST_FILE);
-
-function isFile(res) {
-  try {
-    return fs.statSync(res).isFile();
-  } catch (e) {
-    return false;
-  }
-}
-
-function isDirectory(res) {
-  try {
-    return fs.statSync(res).isDirectory();
-  } catch (e) {
-    return false;
-  }
-}
-
-function readDir(dir) {
-  return fs.statSync(dir).isDirectory() ?
-      Array.prototype.concat(...fs.readdirSync(dir).map(f => readDir(path.join(dir, f)))) :
-      dir.replace(/\\/g, '/');
-}
-
-function loadAsFileSync(res) {
-  if (isFile(res)) {
-    return res;
-  }
-  if (isFile(res + '.js')) {
-    return res;
-  }
-  return null;
-}
-
-function loadAsDirectorySync(res) {
-  const pkgfile = path.join(res, 'package.json');
-  if (isFile(pkgfile)) {
-    try {
-      const pkg = JSON.parse(fs.readFileSync(pkgfile, 'UTF-8'));
-      const main = pkg['main'];
-      if (main) {
-        if (main === '.' || main === './') {
-          main = 'index';
-        }
-
-        let maybe = loadAsFileSync(path.resolve(res, main));
-        if (maybe) {
-          return maybe;
-        }
-
-        maybe = loadAsDirectorySync(path.resolve(res, main));
-        if (maybe) {
-          return maybe;
-        }
-      }
-    } catch (e) {
-    }
-  }
-  return loadAsFileSync(path.resolve(res, 'index'));
-}
-
-function resolveManifestFile(res) {
-  const maybe = runfilesManifest[res] || runfilesManifest[res + '.js'];
-  if (maybe) {
-    return maybe;
-  }
-  // Look for tree artifacts that match and update
-  // the runfiles with files that are in the tree artifact.
-  // Attempt to resolve again with the updated runfiles
-  // if a tree artifact matched.
-  let segments = res.split('/');
-  segments.pop();
-  while (segments.length) {
-    const test = segments.join('/');
-    const tree = runfilesManifest[test];
-    if (tree && isDirectory(tree)) {
-      // We have a tree artifact that matches
-      const files = readDir(tree).map(f => path.relative(tree, f).replace(/\\/g, '/'));
-      files.forEach(f => {
-        runfilesManifest[path.posix.join(test, f)] = path.posix.join(tree, f);
-      })
-      return runfilesManifest[res] || runfilesManifest[res + '.js'];
-    }
-    segments.pop();
-  }
-}
+// we patch require here
+module.constructor._load(patchModulePath)
+// then we also patch process.execArgs sothat any child process also does this patch
+process.execArgv = process.execArgv.concat([`--require=${patchModulePath}`]);
```

Comment:
Will nodejs will already have a `--require=...loader.js` arg in `execArgv`? 
So when a child process or worker executes it again i would expect this logic to run node with `node --require=loader.js --require=patcher.js`  and add another `--require=patcher.js
`
the extra --require=...patcher.js are a no op because loader requires it but the failure mode here is using up the processes argv space unexpectedly.

Reply:
Yeah, this isn't right at all, I was just testing what would happen if you modify the execArgv from inside the program - turns out it's ignored by node in the child process (maybe it should be marked as `readonly` then?)
Either way, we'll move this into the `node_launcher.sh` script

Reply:
migrated to the `--require` from launcher implementation, so we don't need this anymore